lte/gateway/c/session_manager/SessionStateEnforcer.h

```diff
@@ -159,6 +159,23 @@ class SessionStateEnforcer {
       std::unique_ptr<SessionState>& session, SessionFsmState target_state,
       SessionStateUpdateCriteria* session_uc);
 
+  /*
+   * Acquire and update session rules based on the IMSI
+   *
+   * */
+  uint32_t update_session_rules(
+      std::unique_ptr<SessionState>& session, bool gnb_teid_get,
+      bool upf_teid_get, SessionStateUpdateCriteria* session_uc);
+
+  /* update the GNB endpoint details in a rule */
+  bool insert_pdr_from_core(
+      const std::string& imsi, std::unique_ptr<SessionState>& session,

```

Comment:
the imsi param is not needed here. Can do session->get_imsi()

Reply:
omitted parameter imsi here